zap/src/main/java/org/parosproxy/paros/core/scanner/HostProcess.java

```diff
@@ -313,12 +314,23 @@ public void run() {
             }
 
             for (StructuralNode startNode : startNodes) {
+                Map<String, Integer> indexOfTemporary = new HashMap<>();
                 traverse(
                         startNode,
                         true,
                         node -> {
                             if (canScanNode(node)) {
-                                messagesIdsToAppScan.add(node.getHistoryReference().getHistoryId());
+                                Integer index = indexOfTemporary.get(nodeHash(node));
+                                int nodeHistoryId = node.getHistoryReference().getHistoryId();
+                                if (index == null) {
```

Comment:
This is not working correctly when the temp node is after the non-temp node, it still scans both of them.

Reply:
I incorrectly assumed they always came first. Thx for the catch